torch_geometric/explain/metrics.py

```diff
@@ -0,0 +1,46 @@
+from typing import Tuple
+
+import torch
+from torchmetrics import AUROC, ROC
+
+from torch_geometric.explain import Explanation
+
+
+def groundtruth_metrics(
+        explanation: Explanation,
+        groundtruth: Explanation) -> Tuple[float, float, float, float, float]:
+    """accuracy_scores: Compute accuracy scores when
+    groundtruth is available
+
+    Args:
+        explanation (Explanation): Explanation output generated by an explainer
+        groundtruth (Explanation): Groundtruth for explanation
+    """
+
+    ex_masks = explanation.masks
+    ex_mask_tensor = torch.cat(
+        list(map(lambda x: x.view(-1), ex_masks.values())))
+    gt_masks = groundtruth.masks
+    gt_mask_tensor = torch.cat(
+        list(map(lambda x: x.view(-1), gt_masks.values())))
+    threshold = 0
+    ex_mask_tensor[ex_mask_tensor > threshold] = 1.0
+    gt_mask_tensor[gt_mask_tensor > threshold] = 1.0
+
+    correct_preds = gt_mask_tensor == ex_mask_tensor
+    incorrect_preds = gt_mask_tensor != ex_mask_tensor
+    tp = torch.sum(gt_mask_tensor[correct_preds])
+    tn = torch.sum(correct_preds) - tp
+    fp = torch.sum(ex_mask_tensor[incorrect_preds])
+    fn = torch.sum(gt_mask_tensor[incorrect_preds])
+
+    precision = tp / (tp + fp)
+    accuracy = (tp + tn) / (tp + fp + tn + fn)
+    recall = tp / (tp + fn)
+    f1_score = 2 * (precision * recall) / (precision + recall)
+    roc = ROC(task="binary")
```

Comment:
Don't you need to do this before you threshold out the `ex_mask_tensor`? Because here `roc` receives a binary tensor, which it cannot threshold?

Reply:
Yes. That's right. I have updated the implementation to use original ``ex_mask_tensor`` in ROC.